### DISCUSSION TITLE
get dimensions of area

### DISCUSSION THREAD

**Discussion by nschloe** [Textualize/textual#161] [Ideas]
For my [top](https://en.wikipedia.org/wiki/Top_(software))-equivalent in Textual, I draw a bunch of graphs in areas:

![screenshot](https://user-images.githubusercontent.com/181628/141455554-74d1b253-150b-4c8d-8738-e453de963cf1.png)

```python
    class TiptopApp(App):
        async def on_mount(self) -> None:
            grid = await self.view.dock_grid(edge="left")

            # 8/13: approx golden ratio :)
            grid.add_column(fraction=8, name="left")
            grid.add_column(fraction=13, name="right")

            grid.add_row(size=1, name="topline")
            grid.add_row(fraction=1, name="top")
            grid.add_row(fraction=1, name="center")
            grid.add_row(fraction=1, name="bottom")

            grid.add_areas(
                area0="left-start|right-end,topline",
                area1="left-start|right-end,top",
                area2="left,center",
                area3="left,bottom",
                area4="right,center-start|bottom-end",
            )

            grid.place(
                area0=InfoLine(),
                area1=CPU(),
                area2=Mem(),
                area3=Net(),
                area4=ProcsList(),
            )
```
In the widgets that generate the graph, I would like to have some way of knowing the dimensionality of the area such that I can scale the graphs accordingly. I'd also like to have some "resize" event that I can act upon.

Is this available yet?

**Comment by willmcgugan** [comment]
That looks great!

There is a [resize](https://github.com/willmcgugan/textual/blob/main/src/textual/events.py#L86) event, which will give you the new dimensions of the widget.

**Reply by nschloe** [reply]
Is this information available `on_mount`, too?

**Reply by willmcgugan** [reply]
It's not. The mount event happens prior to the layout, but you will get a resize event immediately after (but before rendering).

**Reply by nschloe** [reply]
> but before rendering

That's good enough. Thanks!

**Comment by rudyvan** [comment]
Hi,
I was wondering where these functions InfoLine, CPU, Mem, Net and ProcsList are coming from?
Is there a standard module where they can be imported from?
Or have you made them yourself from modules such as platform?
Thanks for your reply.

**Reply by nschloe** [reply]
> Or have you made them yourself from modules such as platform?

An psutils, yes.

**Comment by rudyvan** [comment]
thanks.
Should you ever want to share this code, i am interested!
